### ISSUE TITLE
Incorrect Type Alias picked in Compilation Error Message

### ISSUE THREAD

**Issue by mmohit-bc** [rust-lang/rust#138055]
### Code

```Rust
So, I have a 

`create(filename).expect(format!("This failed {}", &filename));`

and somewhere I had declared `pub use String as ExecId;` in some module somewhere.

As a result, the error I am getting misleading compilation error message:

I understand `ExecId` is also String alias but it is unrelated thing.
```

### Current output

```Shell
error[E0308]: mismatched types
   --> services/src/abc.rs:218:25
    |
218 |                   .expect(format!("This failed {}",
    |  _________________________^
219 | |                                             &filename));
    | |_____________________________________________________________________^ expected `&str`, found `ExecId`
    |
```

### Desired output

```Shell
error[E0308]: mismatched types
   --> services/src/abc.rs:218:25
    |
218 |                   .expect(format!("This failed {}",
    |  _________________________^
219 | |                                             &filename));
    | |_____________________________________________________________________^ expected `&str`, found `String`
    |
```

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.83.0 (90b35a623 2024-11-26) (Homebrew)
binary: rustc
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf
commit-date: 2024-11-26
host: aarch64-apple-darwin
release: 1.83.0
LLVM version: 19.1.6
```

**Comment by bjorn3** [comment]
`use Foo as Bar;` is not a type alias. `type Foo = Bar;` would be a type alias. `use Foo as Bar;` is just importing an item under a different name. Generally when doing so using the original name would be more confusing that using the name under which you imported it.

**Comment by Marcono1234** [comment]
Similar to / same as #81973? (would be nice if that could be reopened, see my comment)
Not sure if there are more issues about this as well.

----

> Generally when doing so using the original name would be more confusing that using the name under which you imported it.

It seems these re-exports are picked up by the error message even if they are not directly referred to in the current file, or worse they might even come from a third-party crate, and the error message also gives no hint that this is a re-exported type (or where it is defined). See also https://github.com/rust-lang/rust/issues/81973#issuecomment-4957307110.
